minhphan156/book-store
# Implemented book cover upload functionality and integration with AWS S3

![header](https://user-images.githubusercontent.com/34418628/56463903-b552fa80-6392-11e9-8234-23f8a8420c4c.PNG)
![body](https://user-images.githubusercontent.com/34418628/56463904-b97f1800-6392-11e9-9489-9ba49465002d.PNG)

Starting the backend:
First, do npm install on your backend server to install the new multer dependency.
Then, start your backend server with npm start.

To test functionality with Postman:
Have an image file locally ready for upload
Use Postman's POST method using form-data (NOT x-www-form-urlencoded)
In Header, set Content-Type to multipart/form-data
Fill out the body with author, genre, isbn, title, quantity, price
Add a field cover to the body and make sure it is of type file (on the right side of the "KEY" column in Postman you should be able to change it to File)
When you change cover to a File, you will be able to submit a file for upload
Post with Postman to http://localhost:3000/users/upload - your frontend should receive a successful response message and your backend should log successful upload and database manipulation
You can use your AWS S3 credentials to check that the S3 received the image, and you can use PGAdmin to check that the database was updated correctly
